src/main/java/uk/ac/ebi/ampt2d/accession/service/MonotonicAccessionGenerator.java

```diff
@@ -0,0 +1,241 @@
+/*
+ *
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.service;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.data.util.Pair;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.ac.ebi.ampt2d.accession.serial.block.MonotonicRange;
+import uk.ac.ebi.ampt2d.accession.serial.block.MonotonicRangePriorityQueue;
+import uk.ac.ebi.ampt2d.accession.serial.block.persistence.entities.ContiguousIdBlock;
+import uk.ac.ebi.ampt2d.accession.serial.block.persistence.repositories.ContiguousIdBlockRepository;
+import uk.ac.ebi.ampt2d.accession.service.exceptions.AccessionIsNotPending;
+import uk.ac.ebi.ampt2d.accession.utils.ExponentialBackOff;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class MonotonicAccessionGenerator implements InitializingBean {
```

Comment:
Add class-level documentation about usage limitations: a single, non-committed request should not contain say 10 million objects because the system will run out of memory. These requests should be made in batch.